pkg/controller/rds/dbinstance/setup.go

```diff
@@ -0,0 +1,373 @@
+	in.PerformanceInsightsRetentionPeriod = aws.LateInitializeInt64Ptr(in.PerformanceInsightsRetentionPeriod, db.PerformanceInsightsRetentionPeriod)
+	in.PreferredBackupWindow = aws.LateInitializeStringPtr(in.PreferredBackupWindow, db.PreferredBackupWindow)
+	in.PreferredMaintenanceWindow = aws.LateInitializeStringPtr(in.PreferredMaintenanceWindow, db.PreferredMaintenanceWindow)
+	in.PromotionTier = aws.LateInitializeInt64Ptr(in.PromotionTier, db.PromotionTier)
+	in.PubliclyAccessible = aws.LateInitializeBoolPtr(in.PubliclyAccessible, db.PubliclyAccessible)
+	in.StorageEncrypted = aws.LateInitializeBoolPtr(in.StorageEncrypted, db.StorageEncrypted)
+	in.StorageType = aws.LateInitializeStringPtr(in.StorageType, db.StorageType)
+	in.Timezone = aws.LateInitializeStringPtr(in.Timezone, db.Timezone)
+
+	if db.Endpoint != nil {
+		in.Port = aws.LateInitializeInt64Ptr(in.Port, db.Endpoint.Port)
+	}
+
+	if len(in.DBSecurityGroups) == 0 && len(db.DBSecurityGroups) != 0 {
+		in.DBSecurityGroups = make([]string, len(db.DBSecurityGroups))
+		for i, val := range db.DBSecurityGroups {
+			in.DBSecurityGroups[i] = aws.StringValue(val.DBSecurityGroupName)
+		}
+	}
+	if aws.StringValue(in.DBSubnetGroupName) == "" && db.DBSubnetGroup != nil {
+		in.DBSubnetGroupName = db.DBSubnetGroup.DBSubnetGroupName
+	}
+	if len(in.EnableCloudwatchLogsExports) == 0 && len(db.EnabledCloudwatchLogsExports) != 0 {
+		in.EnableCloudwatchLogsExports = db.EnabledCloudwatchLogsExports
+	}
+	if len(in.ProcessorFeatures) == 0 && len(db.ProcessorFeatures) != 0 {
+		in.ProcessorFeatures = make([]*svcapitypes.ProcessorFeature, len(db.ProcessorFeatures))
+		for i, val := range db.ProcessorFeatures {
+			in.ProcessorFeatures[i] = &svcapitypes.ProcessorFeature{
+				Name:  val.Name,
+				Value: val.Value,
+			}
+		}
+	}
+	if len(in.VPCSecurityGroupIDs) == 0 && len(db.VpcSecurityGroups) != 0 {
+		in.VPCSecurityGroupIDs = make([]string, len(db.VpcSecurityGroups))
+		for i, val := range db.VpcSecurityGroups {
+			in.VPCSecurityGroupIDs[i] = aws.StringValue(val.VpcSecurityGroupId)
+		}
+	}
+	in.EngineVersion = aws.LateInitializeStringPtr(in.EngineVersion, db.EngineVersion)
+	// When version 5.6 is chosen, AWS creates 5.6.41 and that's totally valid.
+	// But we detect as if we need to update it all the time. Here, we assign
+	// the actual full version to our spec to avoid unnecessary update signals.
+	if strings.HasPrefix(aws.StringValue(db.EngineVersion), aws.StringValue(in.EngineVersion)) {
+		in.EngineVersion = db.EngineVersion
+	}
+	if in.DBParameterGroupName == nil {
+		for i := range db.DBParameterGroups {
+			if db.DBParameterGroups[i].DBParameterGroupName != nil {
+				in.DBParameterGroupName = db.DBParameterGroups[i].DBParameterGroupName
+				break
+			}
+		}
+	}
+
+	return nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.DBInstance, out *svcsdk.DescribeDBInstancesOutput) (bool, error) {
+	db := out.DBInstances[0]
```

Comment:
One of the things we need to port over from `RDSInstance` is to [skip updating](https://github.com/crossplane/provider-aws/blob/5a380ad/pkg/controller/database/rdsinstance.go#L189) if a modification is on its way and it seems like this function could be the best place for now to do it.